### PR TITLE
Search Component

### PR DESCRIPTION
**Pull request by abpwrs**
### Description: 
Closes issue #51

### Notes:
This component is a simple search over the first and last name of a patient using an exact matching scheme. The next iteration of this functionality should make the search more robust (fuzzy matching).

### Screenshots: 
(You may have to click on the image to view it well)

#### Initial View
| Desktop | Mobile |
| - | - |
|Screen Shot 2020-04-27 at 22 14 44|Screen Shot 2020-04-27 at 22 20 28|

#### Search w/ multiple hits
| Desktop | Mobile |
| - | - |
| Screen Shot 2020-04-27 at 22 15 47| Screen Shot 2020-04-27 at 22 18 41 |

#### Search w/ single hit
| Desktop | Mobile |
| - | - |
|Screen Shot 2020-04-27 at 22 15 57 | Screen Shot 2020-04-27 at 22 19 13|

#### Search w/ no hits
| Desktop | Mobile |
| - | - |
| Screen Shot 2020-04-27 at 22 16 22 | Screen Shot 2020-04-27 at 22 17 37 |

### Checklist:
- [x] 90% Test Coverage (100%)

### CODE REVIEW

File: pwa/src/components/PatientSearch.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    let addNewPatient = () => {
+        console.log('NO PATIENT FOUND, ADDING A NEW ONE')
+    }
+
+    let navigateToPatientPage = (patient: Patient) => {
+        console.log(`Navigating to patient ${JSON.stringify(patient)}`)
+    }
+
+    return (
+        <div>
```
Comment (bmitchinson) [inline pwa/src/components/PatientSearch.tsx:50]: Replace outer `<div>` + `</div>` with `<>` + `</>` ([Fragments](https://reactjs.org/docs/fragments.html))

File: pwa/src/components/PatientSearch.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            <SearchBar
+                data-testid="search-bar"
+                placeholder="Patient Name"
+                maxLength={30}
+                cancelText="Cancel"
+                onChange={(value: string) => searchForName(value)}
+            />
+            <WhiteSpace />
+            {searchValue ? (
+                <div>
```
Comment (bmitchinson) [inline pwa/src/components/PatientSearch.tsx:60]: Replace outer `<div>` + `</div>` with `<>` + `</>` ([Fragments](https://reactjs.org/docs/fragments.html))

File: pwa/src/components/LandingPage.test.tsx [hunk 3]
```diff
@@ -1,11 +1,17 @@
...
+
 import LandingPage from './LandingPage'
 
-test('renders learn react link', () => {
-    const { getByText } = render(<LandingPage />)
-    const lookupButtonElement = getByText(/Patient Lookup/i)
-    const newPatientButtonElement = getByText(/New Patient/i)
+test('renders landing buttons', () => {
+    const landingComponent = render(
+        <Router>
```
Comment (bmitchinson) [inline pwa/src/components/LandingPage.test.tsx:9]: Thanks for this fix to where useHistory can't be used if not wrapped 👍

File: pwa/src/components/PatientSearch.test.tsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
+import React from 'react'
+import { render, fireEvent } from '@testing-library/react'
+import PatientSearch from './PatientSearch'
+
+// TODO: mock out patient data
+
+let searchComponent: any
+let searchBar: HTMLInputElement
+
+beforeEach(() => {
```
Comment (bmitchinson) [inline pwa/src/components/PatientSearch.test.tsx:10]: Good use of beforeEach + afterEach

**Comment by bmitchinson** [comment]
image

X button to clear input isn't functional. 

Good to approve after the above comments, and if we could do any of the following:
- remove it from displaying with some display none css
- enable it's functionality
- make a ticket to do it later

File: pwa/src/components/PatientSearch.tsx (hunk 2 above)
Comment (abpwrs) [reply pwa/src/components/PatientSearch.tsx]: fixed in https://github.com/Haiti-Distributed-Healthcare-System/hdhs/pull/95/commits/399c2f29ce9a37499df2e71f6d376851d78effc0

File: pwa/src/components/PatientSearch.tsx (hunk 1 above)
Comment (abpwrs) [reply pwa/src/components/PatientSearch.tsx]: fixed in https://github.com/Haiti-Distributed-Healthcare-System/hdhs/pull/95/commits/399c2f29ce9a37499df2e71f6d376851d78effc0

**Comment by abpwrs** [comment]
> image
> 
> X button to clear input isn't functional.
> 
> Good to approve after the above comments, and if we could do any of the following:
> 
> * remove it from displaying with some display none css
> * enable it's functionality
> * make a ticket to do it later

This works better in mobile I believe, but could definitely be removed

**Pull request merged**